src/py/mat3ra/made/tools/build/defect/builders.py

```diff
@@ -67,3 +69,97 @@ class SubstitutionPointDefectBuilder(PointDefectBuilder):
 
 class InterstitialPointDefectBuilder(PointDefectBuilder):
     _generator: PymatgenInterstitial = PymatgenInterstitial
+
+
+class SlabDefectBuilderParameters(BaseModel):
+    auto_add_vacuum: bool = True
+    vacuum_thickness: float = 5.0
+
+
+class SlabDefectBuilder(BaseBuilder):
+    _BuildParametersType = SlabDefectBuilderParameters
+    _DefaultBuildParameters = SlabDefectBuilderParameters()
+
+
+class AdatomSlabDefectBuilder(SlabDefectBuilder):
+    _ConfigurationType: type(AdatomSlabDefectConfiguration) = AdatomSlabDefectConfiguration  # type: ignore
+    _GeneratedItemType: Material = Material
+
+    def add_adatom(
+        self,
+        material: Material,
+        chemical_element: str = "Si",
+        position_on_surface: List[float] = [0.5, 0.5],
+        distance_z: float = 2.0,
+    ) -> List[Material]:
+        material_copy = material.clone()
+        basis = material_copy.basis
+        distance_in_crystal_units = distance_z / material_copy.lattice.c
+        max_z = get_atomic_coordinates_extremum(material_copy)
+        position = position_on_surface.copy()
+        position.append(max_z + distance_in_crystal_units)
+        basis.add_atom(chemical_element, position)
+        material_copy.basis = basis
+        return [material_copy]
+
+    def _generate(self, configuration: _ConfigurationType) -> List[_GeneratedItemType]:
+        return self.add_adatom(
+            material=configuration.crystal,
+            chemical_element=configuration.chemical_element,
+            position_on_surface=configuration.position_on_surface,
+            distance_z=configuration.distance_z,
+        )
+
+
+class EquidistantAdatomSlabDefectBuilder(SlabDefectBuilder):
+    _ConfigurationType: type(AdatomSlabDefectConfiguration) = AdatomSlabDefectConfiguration  # type: ignore
+    _GeneratedItemType: Material = Material
+
+    def add_adatom_equdistant(
+        self,
+        material: Material,
+        chemical_element: str = "Si",
+        approximate_position_on_surface: List[float] = [0.5, 0.5],
+        distance_z: float = 2.0,
+    ) -> List[Material]:
```

Comment:
This class should inherit from the one above and reuse its create_adatom. A function `get_equidistant_position(approximate_position_on_surface)` should be added